pkgs/ocaml-modules/ppx_protocol_conv/default.nix

```diff
@@ -0,0 +1,43 @@
+{ buildDunePackage
+, fetchFromGitHub
+, lib
+, base
+, ppxlib
+, ppx_fields_conv
+, ppx_sexp_conv
+, sexplib
+, alcotest
+}:
+
+buildDunePackage rec {
+  pname = "ppx_protocol_conv";
+  version = "5.2.1";
+
+  minimumOCamlVersion = "4.08";
+
+  src = fetchFromGitHub {
+    owner = "andersfugmann";
+    repo = "ppx_protocol_conv";
+    rev = "9a1c4f450cf4fc797e09296d736e5c21f6eb7c4a";
+    sha256 = "sha256-3WlpuuRJfZA6UaeuJRuYI2Z9J7Yng+/J09BDmETwy30=";
+  };
+
+  useDune2 = true;
+
+  #nativeBuildInputs = [ ];
+  propagatedBuildInputs = [
```

Comment:
Is `buildInputs` what you are looking for?

Reply:
3dde5d11b

Reply:
So using only `buildInputs` doesn't work?

Reply:
`buildInputs` works, but all the other packages in `pkgs/ocaml-modules` seem to favor `propagatedBuildInputs`, so I assumed that's the "right" target? Should I change the other modules as well then?

Reply:
Hmm ... you are right, also in nixpkgs most OCaml modules are packaged this way.

https://nixos.org/manual/nixpkgs/stable/#sec-language-ocaml-packaging does not shed too much wisdom on it.

I guess propagated is needed for transitive dependencies as the build only really happens at the root of the tree.